purchases/index.js

```diff
@@ -505,8 +505,16 @@ function renderHtmlReceipt(purchase) {
     return mustache.render(receiptEmailLayout, purchase.data.viewModel);
 }
 
-function listPurchases(db, callback) {
-    db.query("SELECT * FROM purchase", (error, result) => {
+function listPurchases(db, options, callback) {
+    if(!callback){
+        callback = options
+        options = {};
+    }
+   
+    const offset = getOffset(options);
+    const limit = getLimit(options);
+
+    db.query(`SELECT * FROM purchase OFFSET ${offset} LIMIT ${limit}`, (error, result) => {
```

Comment:
You also need to handle th case where `offset` and `limit` are undefined - in that case we shouldn't write `OFFSET` or `LIMIT` (respectively) into the SQL statement at all.